### PR TITLE
Allow crowdfund issues

### PR DESCRIPTION
**Pull request by jvmaia**
##### Description
Allow the crowd to increase contributions to a bounty, so we can crowdfund issues.

##### Checklist

- [X] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines](https://github.com/gitcoinco/web/blob/master/docs/CONTRIBUTING.md#step-4-commit)

##### Affected core subsystem(s)
urls, views

##### Testing
I have tested this as the funder and the collaborator.

##### Refers/Fixes
Fixes: #1380

### CODE REVIEW

**Comment by jvmaia** [comment]
Does the stickler-ci verify when a variable is used inside a f-string?

**Comment by owocki** [comment]
trying to test this and i keep getting this error when i load the 'add contribution' page http://bits.owocki.com/3B2L2R0n172Y/Screen%20Shot%202018-07-02%20at%205.57.34%20PM.png

**Comment by owocki** [comment]
i keep getting crazy high gas fees when i try to use this feature ( screenshot : http://bits.owocki.com/2T1M3e2r3Q3v/Screen%20Shot%202018-07-02%20at%205.58.38%20PM.png )

Usually that means theres a smart contract error on thetx i'm about to submit

**Comment by jvmaia** [comment]
Ook, I'll try to remove the if/else statement inside do_bounty function and separate it into 3 different functions.

**Comment by jvmaia** [comment]
Hey, I didn't got this error here, weird. But I'll make the change because it's more readable.

**Comment by owocki** [comment]
> Hey, I didn't got this error here, weird. But I'll make the change because it's more readable.

Here are the reproduction steps for the issue.

In a browser:

1. Create a new bounty

In another browser with different metamask seed:

1. Open the bounty you created.
2. Click 'Add contribution'.
3. Submit the 'increase funding form'
4. Metamask error appears.

File: app/assets/v2/js/pages/bounty_details.js [hunk 1]
```diff
@@ -662,9 +661,8 @@ var do_actions = function(result) {
     }
 
     if (show_increase_bounty) {
```
Comment (owocki) [inline app/assets/v2/js/pages/bounty_details.js]: won't you need to update `show_increase_bounty` too?

**Comment by owocki** [comment]
debugging further, it seems to call `do_bounty_as_funder_increasingPayout` even when i am not the funder

**Comment by owocki** [comment]
when there is no contract error, i get a [very high gas price](http://bits.owocki.com/0z1n3P1O3U0c/Screen%20Shot%202018-07-03%20at%208.31.17%20AM.png)

**Comment by owocki** [comment]
9dc7a12809aa53652ecddfa47c285cee314fca38 should fix the issues with `is_funder`

**Comment by owocki** [comment]
I'm still seeing contract errors when I try to use the flow though.  I think this just needs to be testeded end to end in each of these scenarios:

```
    if (document.isFunder) {
      if (document.changePayoutAmount) {
        do_bounty_as_funder_increasingPayout();
      } else {
        do_bounty_as_funder_changingFulfillment();
      }
    } else {
      do_bounty_as_contributor();
    }
```

**Comment by jvmaia** [comment]
thanks for the fixes, sorry for the delay, I'll test it now!

File: app/assets/v2/js/pages/bounty_details.js (hunk 1 above)
Comment (jvmaia) [reply app/assets/v2/js/pages/bounty_details.js]: Is it needed? what I change here is the argument from `enabled = increase_bounty_enabled;` to `enabled = true;`, the `show_increase_bounty` is fine.

**Comment by jvmaia** [comment]
> Here are the reproduction steps for the issue.
ook, I did it exactly as you described, but the error does not appears to me. I'm using firefox quantum to submit the 'increase funding form' as the contributor. screenshot: https://i.imgur.com/Lj7Ix9L.png

**Comment by jvmaia** [comment]
my next commits fix the variable `document.changePayoutAmount` and the if/else to decide which function call.

**Comment by jvmaia** [comment]
the error appears only when the `do_bounty_as_funder_changingFulfillment` is called, I'll dig into it tomorrow.

**Comment by jvmaia** [comment]
Ook, I find the problem, the `contract.changeBountyFulfillment` requires the issue is at stage **Draft** ([doc link](https://github.com/Bounties-Network/StandardBounties/blob/master/contracts/StandardBounties.sol#L448)), and the `contract.issueAndActivateBounty`, used to create the bounty, set the issue's stage to **Active** ([doc link](https://github.com/Bounties-Network/StandardBounties/blob/master/contracts/StandardBounties.sol#L228)).

**Comment by owocki** [comment]
oh shoot.. i guess that means that we can't do this after all... ? @jvmaia thats my fault, i can still pay out 

@mbeylin any workarounds for https://github.com/gitcoinco/web/pull/1597#issuecomment-402597308 you can think of ?

**Comment by jvmaia** [comment]
We can deploy a support contract to call **transitionToState()** to define issue's stage to *Draft*, call the **changeBountyfulfillmentAmount()**, and call the support contract again to go back to issue's stage *Active*. But i don't know if it would be the best workaround.

**Comment by owocki** [comment]
very sorry we could not merge this..

@mbeylin let us know if you see any potential workarounds for 1.0.. this is a highly requested feature from our community

**Comment by owocki** [comment]
unfortunately it looks liek `transitionToState()` is an internal only method right now

**Comment by jvmaia** [comment]
ooh sorry @owocki, it would be a pleasure to work on the workaround to this.

**Pull request closed without merging**